### DISCUSSION TITLE
VS Code asks to login to my GitHub Copilot everytime I open VS Code

### DISCUSSION THREAD

**Discussion by saanville** [community/community#18228] [Copilot Conversations] [answered]
Every time I open VS Code on Mac, it always asks me to login to GitHub Copilot using GitHub. My GitHub uses Two-Factor Authentication to login which makes the whole process take up a lot of time.

**Comment by MrKrishnaAgarwal** [comment]
Update your VS Code
I also use Two-Factor Authentication
but never faced this issue

**Reply by jacobgraf** [reply]
I am having this same issue and it's maddening...

VS Code: 1.69.0-insider
Copilot: 1.31.6194

Screenshot: https://cln.sh/wIAEs1XbiBfsaUWRrBwN

**Comment by davecheney** [comment]
@saanville @jacobgraf Take it from someone who works here, 2fa is not the issue 😉 
<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.

**Reply by jacobgraf** [reply]
Yep. Did that multiple times. Same issue. I also completely uninstalled VS Code and reinstalled. Same issue. I understand 2FA isn't the issue, but what is? It's prompting me to login every time I start the app. Also, I see two notifications down in the bottom left telling me to login again for settings sync and copilot. If I close and reopen VS Code, those are back to wanting me to login again. It's maddening. How do I get my logins to stick?

**Reply by davecheney** [reply]
Have you tried the non insiders build? Please double check that your using VScode, not VScodum or OSS Code

**Reply by jacobgraf** [reply]
Yes. Same issue. Definitely using VS Code too. It feels like it has to be some discrepancy between VS Code and macOS Keychain or something.

![image](https://user-images.githubusercontent.com/146333/176508046-c1c70ffa-5a96-4f00-8abf-9be4a85b5947.png)

**Reply by jacobgraf** [reply]
I am on macOS Ventura B2, so maybe there's an issue there too. I just searched "github" in Keychain Access and deleted everything, then signed in to VS Code again and then quit and reloaded it. Same issue.

**Reply by davecheney** [reply]
@jacobgraf looks like a firewall issue. Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Reply by jacobgraf** [reply]
No? How do you know it's a firewall issue tho?? If it was, wouldn't it not allow you to login at all? I can log in to CoPilot and Settings Sync using GitHub and it works just fine, but the problem is that after I close and reopen VS Code, it's asking me to do it again.

**Reply by davecheney** [reply]
From the limited telemetry your editor is emitting, I can see that some hosts are not contactable from your machine. It looks like DNS lookups are failing. I cannot be more specific for privacy reasons. The short version is, you're asked to log in all the time because login does not succeed. 

Can you please paste the relevant parts of View -> Output; select _GitHub Copilot_ from the pulldown.

**Reply by jacobgraf** [reply]
Nothing shows up under CoPilot, but this shows up under GitHub Auth.

![CleanShot 2022-06-29 at 13 44 28@2x](https://user-images.githubusercontent.com/146333/176512418-87915a42-17d6-4cf1-be99-45babad78514.jpg)

**Reply by darknoon** [reply]
Having the exact same issue. Ventura beta 2. Tried signing out / back in, rebooting, etc.

output after restarting
```
[Info  - 15:58:06.307] Reading sessions from keychain...
[Info  - 15:58:06.315] Getting sessions for user:email...
[Info  - 15:58:06.326] Getting sessions for all scopes...
[Info  - 15:58:06.475] Got 0 sessions for user:email...
[Info  - 15:58:06.475] Got 0 sessions for ...
[Info  - 15:58:06.716] Getting sessions for read:user...
[Info  - 15:58:06.716] Getting sessions for all scopes...
[Info  - 15:58:06.716] Got 0 sessions for read:user...
[Info  - 15:58:06.716] Got 0 sessions for ...
[Info  - 15:58:06.737] Getting sessions for read:user,repo,user:email...
[Info  - 15:58:06.737] Got 0 sessions for read:user,repo,user:email...
```

**Comment by JaimeTR** [comment]
I installed GitHub Copilot Nightly, and it fixed the problem for me.
Before installing GitHub Copilot Nightly, please uninstall or disable Github copilot and install the extension from https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-nightly

If this answer is helpful, please 'Mark as answer'. :)

**Reply by darknoon** [reply]
I tried this, and it didn't improve things for me. I disabled the normal plug-in. Not sure why.

**Reply by jacobgraf** [reply]
Same. Tried Nightly and it did not fix it for me either. This is maddening.

**Comment by kgautamt** [comment]
Same issue here. 

VSCode
```
Version: 1.68.1
Commit: 30d9c6cd9483b2cc586687151bcbcd635f373630
Date: 2022-06-14T12:48:37.209Z
Electron: 17.4.7
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Darwin arm64 22.0.0
```

```
Name: GitHub Copilot
Id: GitHub.copilot
Description: Your AI pair programmer
Version: 1.31.6194
Publisher: GitHub
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=GitHub.copilot
```

**Comment by tsjason** [comment]
I am also experiencing this issue. I have to sign in every time I open a new window. I'm using Copilot Nightly and VS Code 1.68.1.

![CleanShot 2022-07-07 at 18 45 45](https://user-images.githubusercontent.com/26229182/177816138-ed2158e9-e3fb-4723-a66b-cbecfca809dd.gif)

**Comment by okisdev** [comment, marked as answer]
Hello everyone. In most cases, when you are experiencing this issue, it is because you are using the macOS 13 Ventura Beta. The beta version changes the authentication process for keychains. This prevents most plugins on VS Code from saving the token, please check this [issue](https://github.com/microsoft/vscode/issues/152964) to learn more.

## Solution
Download the latest VS Code insider version [here](https://code.visualstudio.com/insiders/) (v1.70.0-insider). It should be fixed.

**Reply by i-am-the-slime** [reply]
Lovely, I just got mad enough to google this and here's your answer from 3h ago.

**Reply by agulencina96** [reply]
I have the same issue. Im in Manjaro 21 KDE using the snap version of vscode.
Copilot is working but I got the message every time I open the app.

**Reply by alcidesbsilvaneto** [reply]
Yep, that was it to me. Thank you.

**Reply by jacobgraf** [reply]
Fixed for me too.

**Reply by duraanali** [reply]
This fixed for me! Thanks.

**Reply by sljzdotcom** [reply]
Same problem. Now it's fixed for me, thanks.
